### PR TITLE
Numark DJ2Go2: support HotCue clear with Pad

### PR DESCRIPTION
**Pull request by doteya**
push "Pad" in "CUES" mode, activate hotcue, and
push "Pad" holding "PAD MODE" in "CUES" mode, clear hotcue.

### CODE REVIEW

File: res/controllers/Numark_DJ2GO2_Touch.midi.xml [hunk 1]
```diff
@@ -221,6 +221,78 @@
           <script-binding/>
         </options>
       </control>
+      <control>
```
Comment (doteya) [inline res/controllers/Numark_DJ2GO2_Touch.midi.xml:224]: DJ2GO2/DJ2GO2 Touch does not support "shift" button, but send another control (control + 8) when push "Pad" holding "PAD MODE" in "CUES" mode.  
Other DJ software assigns it to clear hotcue.  
In other modes, holding "PAD MODE" is ignored. DJ2GO2/DJ2GO2 Touch sends same control holding "PAD MODE" or not.

File: res/controllers/Numark_DJ2GO2_Touch_scripts.js [hunk 2]
```diff
@@ -104,7 +104,14 @@ DJ2GO2Touch.Deck = function(deckNumbers, midiChannel) {
         this.hotcueButtons[i] = new components.HotcueButton({
             group: "[Channel" + script.deckFromGroup(this.currentDeck) + "]",
             midi: [0x94 + midiChannel, 0x00 + i],
-            number: i
+            number: i,
+            input: function(channel, control, value, status, _group) {
```
Comment (doteya) [inline res/controllers/Numark_DJ2GO2_Touch_scripts.js:108]: If the received control is smaller than or equal to 8, send _activate to hotcue.
If the received control is greater than 8, send _clear to hotcue.

**Comment by ronso0** [comment]
Thank you for this contribution!
For your work to be included in Mixxx we need you to sign the contributor agreement:
Since this is your first contribution we need you to sign the [contributor agreement](https://docs.google.com/forms/d/e/1FAIpQLScC9QG327sjLL0eWftmfGUasxFFLxg6LCXJ2xHDYRzFIRqyiw/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ) and thereby allows Mixxx to include your changes.
Please give a headsup if you did so.

Updates/fixes to controller mappings that do not rely on new features should go to the current stable branch, which is `2.3` atm.
Please change the base branch here, rebase your branch locally and force-push.

In order to make CI happy (without too many fixup commits), please also set up `pre-commit` locally, see https://github.com/mixxxdj/mixxx/wiki/Using-git#set-up-automatic-code-checking

File: res/controllers/Numark_DJ2GO2_Touch_scripts.js [hunk 3]
```diff
@@ -104,7 +104,14 @@ DJ2GO2Touch.Deck = function(deckNumbers, midiChannel) {
...
+                if (control <= 8) {
```
Comment (Swiftb0y) [inline res/controllers/Numark_DJ2GO2_Touch_scripts.js:109]: From your comment above
```suggestion 
                // DJ2GO2/DJ2GO2 Touch does not have a "shift" button, 
                // but when in CUES Pad Mode, holding the PAD MODE button will 
                // cause the controller to send messages where the originating midi
                // control is offset by 8.
                // this "feature" only applies to the CUES PAD MODE. Holding the button
                // in other modes does nothing.
                if (control <= 8) {
```

**Review by Swiftb0y — CHANGES_REQUESTED** [review]
Thanks for your contribution. Can you also take care of updating the manual with this new feature?

**Comment by doteya** [comment]
Thank you for comments.

ronso0,

I sent my sign.
But I don't know the name is my account name or my real name.
I sent the sign of my real name and my account name with same mail address.

I have branced form 2.3, cherry-pick my commit and push force to my local repositry.

And DJ2Go is not compatible with DJ2Go2/DJ2Go2 Touch.
So I changed commit title DJ2Go to DJ2Go2.
DJ2Go2 and DJ2Go2 Touch are compatible.
Difference of them is "without Touch" or "with Touch" only.

Swiftb0y,

Thank you.

I will try to update manual.

I add comment.

**Comment by Swiftb0y** [comment]
Thanks. If you take care of the manual, I'll can take care of the failing pre-commit hook before merging.

**Comment by ronso0** [comment]
> But I don't know the name is my account name or my real name.
> I sent the sign of my real name and my account name with same mail address.

Well, depends on with which name/nick you want to be mentioned in the contributor list in the About dialog and on the website? 
I'd simply delete the other one.

**Comment by doteya** [comment]
> > But I don't know the name is my account name or my real name.
> > I sent the sign of my real name and my account name with same mail address.
> 
> Well, depends on with which name/nick you want to be mentioned in the contributor list in the About dialog and on the website? I'd simply delete the other one.

I want to use my account name 'doteya'.
Thank you.

**Comment by doteya** [comment]
> Thanks. If you take care of the manual, I'll can take care of the failing pre-commit hook before merging.

I make PR for the manual.
https://github.com/mixxxdj/manual/pull/506

And I have fixed pre-commit code style issue.
I'm always using windows environment. I'm sorry late to setup the pre-commit chek environment.

**Comment by Swiftb0y** [comment]
No Problem.
You seem to be proficient with git. Can you squash all the changes into a single commit and force push?

**Comment by doteya** [comment]
> No Problem. You seem to be proficient with git. Can you squash all the changes into a single commit and force push?

I squash all commits in a single commit and push it.

**Comment by ronso0** [comment]
> I want to use my account name 'doteya'.

Done. Thank you!

**Review by Swiftb0y — APPROVED** [review]
LGTM. thank you very much.

**Pull request merged**